src/cpp/rtps/history/CacheChangePool.cpp

```diff
@@ -101,7 +101,7 @@ void CacheChangePool::return_cache_to_pool(
     ch->isRead = 0;
     ch->sourceTimestamp.seconds(0);
     ch->sourceTimestamp.fraction(0);
-    ch->num_sent_submessages = 0;
+    ch->writer_info.num_sent_submessages = 0;
```

Comment:
At this level I don't know if it was a CacheChange_t used by a reader or a writer. I make this changes because it is already be there. But I leave the initialization when the writer or reader get the CacheChange_t.